lumicks/pylake/scalebar.py

```diff
@@ -0,0 +1,153 @@
+
+    bars = AuxTransformBox(transform)
+    bar_args = {"ec": color, "lw": barwidth, "fc": None}
+
+    if size_x:
+        bars.add_artist(Rectangle(xy=(0, size_y), width=size_x, height=0, **bar_args))
+
+    if size_y:
+        bars.add_artist(Rectangle(xy=(0, 0), width=0, height=size_y, **bar_args))
+
+    textprops = {"color": color, "fontsize": fontsize}
+    if size_x and label_x:
+        xlabel = TextArea(label_x, textprops=textprops)
+        bars = VPacker(children=[bars, xlabel], align="center", pad=0, sep=separation)
+
+    if size_y and label_y is not None:
+        ylabel = TextArea(label_y, textprops=textprops)
+
+        if size_x and label_x is not None:
+            # Add a dummy label with the same text such that the label is properly centered.
+            # Without this dummy element, the label ends up in the middle of the full vertical
+            # height of the scales and the text underneath.
+            dummy = TextArea(label_x, textprops={**textprops, "visible": False})
+            ylabel = VPacker(children=[ylabel, dummy], align="center", pad=0, sep=0)
+
+        bars = HPacker(children=[ylabel, bars], align="center", pad=0, sep=separation)
+
+    return AnchoredOffsetbox(loc, child=bars, frameon=False, **kwargs)
+
+
+@dataclass
+class ScaleBar:
+    """Draws a scale legend for the current axis.
+
+    Parameters
+    ----------
+    size_x, size_y : float
+        Width of the scale bars in data units. `0` or `None` omits the scale.
+    label_x, label_y : str or None
+        Labels for the scale bars. None results in no label on the scale.
+    loc : position in the containing axis
+        Valid locations are 'upper left', 'upper center', 'upper right', 'center left',
+        'center', 'center right', 'lower left', 'lower center', 'lower right'.
+    color : color
+        Color to use for the text and labels.
+    separation : float
+        separation between labels and bars in points.
+    bar_width : float or None
+        Width of the scale bar(s)
+    fontsize : float or None
+        Font size to use for the labels.
+    **kwargs
+        additional arguments passed to :class:`matplotlib.offsetbox.AnchoredOffsetBox`.
+
+    Examples
+    --------
+    ::
+
+        import lumicks.pylake as lk
+
+        # Loading a kymograph.
+        h5_file = pylake.File("example.h5")
+        _, kymo = h5_file.kymos.popitem()
+
+        # Show default scale bar
+        kymo.plot("green", scale_bar=lk.ScaleBar())
+        plt.show()
+
+        # Show scale bar with a scale of 10 seconds on the x axis
+        kymo.plot("green", scale_bar=lk.ScaleBar(size_x=10))
+        plt.show()
+    """
+
+    size_x: float or None = None
+    size_y: float or None = None
+    label_x: str or None = None
+    label_y: str or None = None
+    loc: str or int = "upper right"
+    color: str = "white"
+    separation: float or None = 2.0
+    barwidth: float or None = None
+    fontsize: float or None = None
```

Comment:
`float or None` is not actually a valid type annotation

you can use either `Optional[str]` or `str | None` (Python 3.10+)

Reply:
Good catch, thanks! We still support older versions than 3.10, so I'll go with `Optional[]` and `Union[]`.